src/chai-form-snippet.ts

```diff
@@ -0,0 +1,117 @@
+import { LitElement, html, css } from 'lit';
+import { property, customElement } from 'lit/decorators.js';
+import { ChaiForm } from './chai-form';
+
+@customElement('chai-form-snippet')
+export class ChaiFormSnippet extends LitElement {
+  @property({ type: String }) targetForm: string = '';
+  @property({ type: Boolean }) isCodeSnippedVisible: Boolean = false;
+
+  static override styles = css`
+  
+  .snippet{
+    position: relative;
+
+  }
+
+  .snippet-input{
+    width: 400px;
+    height: 201px;
+  }
+
+  .snippet-header{
+       font-size: 19px;
+    background-color: black;
+    padding: 14px;
+    border-radius: 11px;
+    color: yelo;
+    color: yellow;
+    text-shadow: none;
+    font-family: sans-serif;
+  }
+
+  .snippet-header-message
+  {
+    font-size:16px;
+  }
+
+  .snippet-copy-btn{
+        background-color: black;
+    border-radius: 6px;
+    padding:10px;
+    margin-top: 9px;
+    color: yellow;
+    font-family: sans-serif;
+  }
+  `;
+
+  constructor() {
+    super();
+    this.targetForm = ''; 
+  }
+
+  @property({ type: String }) selectedVisibilityValue: string = '';
+  @property({ type: String }) htmlSnippet: string = '';
+
+
+
+
+
+  getSnippet()
+  {
+    
+      const target = document.getElementById(this.targetForm) as ChaiForm; 
+        if (!target) return;
+  
+    // const snippet= target.shadowRoot?.innerHTML  || '' as string; // Get the inner HTML
+
+    const snippet = `
+${target.outerHTML.replace('</chia-form>', '')}
+${target.shadowRoot?.getElementById('chai-name')?.outerHTML}
+${target.shadowRoot?.getElementById('chai-phone')?.outerHTML}
+${target.shadowRoot?.getElementById('chai-email')?.outerHTML}
+${target.shadowRoot?.getElementById('chai-address')?.outerHTML}
+</chia-form>
+`;
+    console.log(snippet);
+
+
+     try {
+      // Set focus on the document
+      document.body.focus(); // Ensure the document is focused
+
+      // Use setTimeout to allow the browser to focus before copying
```

Comment:
Have you tried this in Safari? Safari doesn't allow delays before interacting with the `navigator.clipboard` API.